rm/st/wlock_helper.c

```diff
@@ -186,8 +186,8 @@ static void _write_lock_put(struct wlock_ctx *wlx)
 M0_INTERNAL void rm_write_lock_put()
 {
 	_write_lock_put(&wlx);
-	wlock_ctx_destroy(&wlx);
 	wlock_ctx_disconnect(&wlx);
+	wlock_ctx_destroy(&wlx);
```

Comment:
That's not enough. It is possible that an RM fom is executed concurrently with rm_write_lock_put() and nothing prevents it from attempting to access the context after it has been finalised.

Reply:
This is something similar to https://github.com/Seagate/cortx-motr/pull/1821  ?

Reply:
This is in the UT/ST code. Do you think we need to fix it now?

Reply:
I propose to add a delay before the wlock_ctx_disconnect() so that will give the user a chance to finish using the wlx.

This is not a thorough fix, but it is in UT/ST code, and maybe we can live with it.